Source/WebCore/css/CSSCounterStyle.cpp

```diff
@@ -27,52 +27,211 @@
 #include "CSSCounterStyle.h"
 
 #include "CSSCounterStyleDescriptors.h"
+#include "CSSCounterStyleRegistry.h"
 #include "CSSCounterStyleRule.h"
 #include "CSSPrimitiveValue.h"
 #include "CSSValuePair.h"
+#include <cmath>
+#include <wtf/text/StringBuilder.h>
+#include <wtf/text/TextBreakIterator.h>
 
 namespace WebCore {
 
-String CSSCounterStyle::counterForSystemCyclic(int)
+// https://www.w3.org/TR/css-counter-styles-3/#cyclic-system
+String CSSCounterStyle::counterForSystemCyclic(int value)
```

Comment:
I forgot to mention, but can we make all of these counterFor* functions `const`?